pkg/subscraping/sources/virustotal/virustotal_test.go

```
@@ -0,0 +1,193 @@
+
+// With no per-source limit the source must follow the cursor to the end and
+// return every subdomain, i.e. the default behaviour stays unbounded so paid
+// keys are never silently truncated.
+func TestVirustotalSource_FollowsCursorWhenUnlimited(t *testing.T) {
+	var hits int32
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		n := atomic.AddInt32(&hits, 1)
+		w.Header().Set("Content-Type", "application/json")
+		// Three pages of one subdomain each, then the cursor terminates.
+		if n < 3 {
+			fmt.Fprintf(w, `{"data":[{"id":"sub%d.example.com"}],"meta":{"cursor":"next-%d"}}`, n, n)
```

golangci-lint (2.12.2)

[error] Error return value of `fmt.Fprintf` is not checked

(errcheck)